modules/wallet/update.go

```diff
@@ -300,6 +297,12 @@ func (w *Wallet) ProcessConsensusChange(cc modules.ConsensusChange) {
 		w.log.Println("ERROR: failed to update consensus change ID:", err)
 	}
 
+	if height, err := dbGetConsensusHeight(w.dbTx); err == nil && height%100 == 0 {
+		// sync every 100 blocks to prevent running transaction from growing
+		// too large
+		w.syncDB()
+	}
```

Comment:
This adds an I/O operation every single block. While not as bad as a sync, it's probably better to just keep a counter in memory.